### PR TITLE
Adds and enables the x86 host runner when building on x86 host!

### PR DESCRIPTION
**Pull request by Sonicadvance1**
Disables the thunking unit test on host because obviously it won't work
there

### CODE REVIEW

File: unittests/ASM/CMakeLists.txt [hunk 1]
```diff
@@ -43,21 +43,28 @@ foreach(ASM_SRC ${ASM_SOURCES})
...
+    "-c irint -n 1"      "int_1"     "int"
+    "-c irint -n 500"    "int_500"   "int"
+    "-c irint -n 500 -m" "int_500_m" "int"
+    "-c irjit -n 1"      "jit_1"     "jit"
+    "-c irjit -n 500"    "jit_500"   "jit"
+    "-c irjit -n 500 -m" "jit_500_m" "jit"
     )
+  if (_M_X86_64)
+    list(APPEND TEST_ARGS
+      "-g -c host"       "host"      "host"
```
Comment (skmp) [inline unittests/ASM/CMakeLists.txt:55]: why not pass -g globally? seems like a good idea anyway

File: unittests/32Bit_ASM/CMakeLists.txt [hunk 2]
```diff
@@ -43,28 +43,31 @@ foreach(ASM_SRC ${ASM_SOURCES})
 
   # Format is "<Test Arguments>" "<Test Name>"
```
Comment (skmp) [inline unittests/32Bit_ASM/CMakeLists.txt:44]: Add "\<Test Runner\>"?

File: unittests/ASM/Disabled_Tests [hunk 3]
```diff
@@ -23,3 +23,6 @@ Test_Primary/Primary_E9.asm
 Test_X87/D9_F9.asm
 
 Test_X87/D9_F2.asm
+
+# 3DNow! no longer exists on AMD Zen CPUs
```
Comment (skmp) [inline unittests/ASM/Disabled_Tests:27]: Now with the `Disabled_Tests_host`, wouldn't it make more sense to move this there?

**Review by skmp — CHANGES_REQUESTED** [review]

File: unittests/ASM/CMakeLists.txt (hunk 1 above)
Comment (Sonicadvance1) [reply unittests/ASM/CMakeLists.txt:55]: Sounds good

File: unittests/32Bit_ASM/CMakeLists.txt (hunk 2 above)
Comment (Sonicadvance1) [reply unittests/32Bit_ASM/CMakeLists.txt:44]: Added

File: unittests/ASM/Disabled_Tests (hunk 3 above)
Comment (Sonicadvance1) [reply unittests/ASM/Disabled_Tests]: Moved

**Review by skmp — APPROVED** [review]

**Pull request merged**